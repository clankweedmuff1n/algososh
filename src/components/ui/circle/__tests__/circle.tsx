import React from 'react';
import { render } from '@testing-library/react';
import { Circle } from '../circle';
import {ElementStates} from "../../../../types/element-states";

describe('Тестирование компонента Circle', () => {
    test('Отрисовка без буквы', () => {
        const { container } = render(<Circle />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка с буквами', () => {
        const { container } = render(<Circle letter="A" />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка с head', () => {
        const { container } = render(<Circle head="Head" />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка с react-элементом в head', () => {
        const { container } = render(<Circle head={<span>Head</span>} />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка с tail', () => {
        const { container } = render(<Circle tail="Tail" />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка с react-элементом в tail', () => {
        const { container } = render(<Circle tail={<span>Tail</span>} />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка с index', () => {
        const { container } = render(<Circle index={1} />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка с пропом isSmall === true', () => {
        const { container } = render(<Circle isSmall />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка в состоянии default', () => {
        const { container } = render(<Circle state={ElementStates.Default} />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка в состоянии changing', () => {
        const { container } = render(<Circle state={ElementStates.Changing} />);
        expect(container).toMatchSnapshot();
    });

    test('Отрисовка в состоянии modified', () => {
        const { container } = render(<Circle state={ElementStates.Modified} />);
        expect(container).toMatchSnapshot();
    });
});
