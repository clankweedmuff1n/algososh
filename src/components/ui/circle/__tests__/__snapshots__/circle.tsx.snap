// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Отрисовка без буквы 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка в состоянии changing 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка в состоянии default 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка в состоянии modified 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка с head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      Head
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка с index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      1
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка с react-элементом в head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <span>
        Head
      </span>
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка с react-элементом в tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <span>
        Tail
      </span>
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка с tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      Tail
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка с буквами 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка с пропом isSmall === true 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;
